service/controller/internal/cloudconfig/tccp.go

```diff
@@ -130,3 +131,287 @@ func (t *TCCP) Render(ctx context.Context, cr infrastructurev1alpha2.AWSCluster,
+				Owner: k8scloudconfig.Owner{
+					Group: k8scloudconfig.Group{
+						Name: FileOwnerGroupName,
+					},
+					User: k8scloudconfig.User{
+						Name: FileOwnerUserName,
+					},
+				},
+				Permissions: 0700,
+			}
+
+			certsMeta = append(certsMeta, meta)
+		}
+	}
+
+	var fileAssets []k8scloudconfig.FileAsset
+
+	data := e.templateDataTCCP()
+
+	for _, fm := range filesMeta {
+		c, err := k8scloudconfig.RenderFileAssetContent(fm.AssetContent, data)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		asset := k8scloudconfig.FileAsset{
+			Metadata: fm,
+			Content:  c,
+		}
+
+		fileAssets = append(fileAssets, asset)
+	}
+
+	for _, cm := range certsMeta {
+		c := base64.StdEncoding.EncodeToString([]byte(cm.AssetContent))
+		asset := k8scloudconfig.FileAsset{
+			Metadata: cm,
+			Content:  c,
+		}
+
+		fileAssets = append(fileAssets, asset)
+	}
+
+	return fileAssets, nil
+}
+
+func (e *MasterExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
+	unitsMeta := []k8scloudconfig.UnitMetadata{
+		// Create symlinks for nvme disks.
+		// This service should be started only on first boot.
+		{
+			AssetContent: cloudconfig.NVMEUdevTriggerUnit,
+			Name:         "ebs-nvme-udev-trigger.service",
+			Enabled:      true,
+		},
+		// Set bigger timeouts for NVME driver.
+		// Workaround for https://github.com/coreos/bugs/issues/2484
+		// TODO issue: https://github.com/giantswarm/giantswarm/issues/4255
+		{
+			AssetContent: cloudconfig.NVMESetTimeoutsUnit,
+			Name:         "nvme-set-timeouts.service",
+			Enabled:      true,
+		},
+		{
+			AssetContent: cloudconfig.DecryptTLSAssetsService,
+			Name:         "decrypt-tls-assets.service",
+			Enabled:      true,
+		},
+		{
+			AssetContent: cloudconfig.DecryptKeysAssetsService,
+			Name:         "decrypt-keys-assets.service",
+			Enabled:      true,
+		},
+		{
+			AssetContent: cloudconfig.SetHostname,
+			Name:         "set-hostname.service",
+			Enabled:      true,
+		},
+		{
+			AssetContent: cloudconfig.EphemeralVarLibDockerMount,
+			Name:         "var-lib-docker.mount",
+			Enabled:      true,
+		},
+		// Mount etcd EBS volume.
+		{
+			AssetContent: cloudconfig.MountEtcdVolume,
+			Name:         "var-lib-etcd.mount",
+			Enabled:      true,
+		},
+		// Mount log EBS volume.
+		{
+			AssetContent: cloudconfig.EphemeralVarLogMount,
+			Name:         "var-log.mount",
+			Enabled:      true,
+		},
+	}
+
+	var newUnits []k8scloudconfig.UnitAsset
+
+	data := e.templateDataTCCP()
+
+	for _, fm := range unitsMeta {
+		c, err := k8scloudconfig.RenderAssetContent(fm.AssetContent, data)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		unitAsset := k8scloudconfig.UnitAsset{
+			Metadata: fm,
+			Content:  c,
+		}
+
+		newUnits = append(newUnits, unitAsset)
+	}
+
+	return newUnits, nil
+}
+
+func (e *MasterExtension) VerbatimSections() []k8scloudconfig.VerbatimSection {
+	newSections := []k8scloudconfig.VerbatimSection{}
+
+	return newSections
+}
```

Comment:
we need different renders for ASG masters and old ec2 instance master so thats why we have this  different for each of them